recipe-server/client/control/components/GroupMenu.js

```diff
@@ -76,32 +104,62 @@ export default class GroupMenu extends React.Component {
     } = this.props;
 
     const textSearchMessage = this.buildTextSearchMessage(searchText);
+    const maxCount = GroupMenu.INTIAL_DISPLAY_COUNT;
 
     return (
       <div
         className="group-menu"
       >
         { textSearchMessage }
         {
-          data.map(group =>
-            <div
-              key={group.value}
-              className={group.value}
-            >
-              <h3 className="group-label">{group.label}</h3>
-              {
-                group.options.map((option, index) =>
-                  <div
-                    className={"menu-item"}
-                    key={index}
-                    onClick={this.makeClickItemHandler(group, option)}
-                  >
-                    { option.label || option.value }
-                  </div>
-                )
-              }
-            </div>
-          )
+          data.map(group => {
+            const {
+              value,
+              options,
+              label,
+            } = group;
```

Comment:
I actually wanted to go with this too, but the event handlers use the `group` object later in that scope, e.g. `...onClick={this.makeClickItemHandler(group, option)}...`.

I can set up the handlers to expect the explicit values from `group`/`option` if you'd prefer. I like to just pass the whole object in so the calling scope doesn't need to know how the handler works.